gpflow/utilities/utilities.py

```diff
@@ -420,6 +420,11 @@ def traverse_module(
             new_acc = (f"{path}['{term_idx}']", new_state)
             new_state = traverse_module(subterm, new_acc, update_cb, target_types)
     elif isinstance(m, tf.Module):
+        items = vars(m)
+        if isinstance(m, tfp.bijectors.Bijector) and "_parameters" in items:
+            assert items["_parameters"]["self"] is m
+            del items["_parameters"]["self"]  # avoid infinite recursion
```

Comment:
This fixes the issue with `leaf_components(bijector)` where `bijector` is any tfp.bijectors.Bijector instance.